apps/documents/views.py

```diff
@@ -88,6 +89,9 @@ def list(self, request, *args, **kwargs):
         else:
             queryset = self.filter_queryset(self.get_queryset())
 
+        for document in queryset:
+            get_document_from_vault.delay(document.id)
```

Comment:
I think repopulating the cache should be synchronous and transparent to the user. In other words, the view should not return until the cache has been repopulated by engine, so I don't think this logic should go in a celery task.

If this turns out to be too slow, we can revisit doing this asynchronous - but we'll need some way to tell the frontend that the image might not actually be in S3 yet (and the frontend will have to implement polling).

You need to also handle cache repopulation in the document `retrieve` view method. Really it needs to happen any time a presigned s3 url is generated for a document, so maybe the cache repopulation logic should go in the serializer?